src/main/java/http_server/ResourceHandler.java

```diff
@@ -58,4 +61,30 @@ public void delete(String resourcePath) {
         );
     }
 
+    public boolean updateJSON(String resourcePath, String patchDocument) {
```

Comment:
I see below that you have a `JSONHandler`, and it makes me wonder why `updateJSON`'s logic sits in the `ResourceHandler`.  What motivated this structure?

Reply:
I had kept the logic in ResourceHandler since it is all about handling a JSON PATCH document and not specific to just JSON handling in general. It made sense in my head to keep JSON Handler to just parse/update/write JSON documents, and to let `resource.updateJSON()` use JSONHandler to do the actual updating per patch instruction. 

If it's not a strong enough reason to keep this logic in ResourceHandler, let me know and i'll move it to JSONHandler!

Reply:
TBH I was just curious about it.  When I see names that sort of blend like in this case, I try to pause and think about the design underlying it.

Like-named systems tend to sort of gravitate towards each other over time, both in terms of code location _and_ in terms of how frequently they change together.  A big knack of software is knowing how to arm yourself for change, and sometimes that means grouping changeable things together.

Buuuuuut, it doesn't always mean you need to reorganize.  Sometimes it just means you need to keep an eye on it.

In this case, I don't think we need to change a whole lot more, and so it can probably be left as-is (unless someone else feels differently).